server/src/main/java/org/opensearch/cluster/routing/RoutingNode.java

```diff
@@ -48,63 +48,140 @@
 import java.util.Map;
 import java.util.Set;
 import java.util.stream.Collectors;
+import java.util.stream.StreamSupport;
 
 /**
  * A {@link RoutingNode} represents a cluster node associated with a single {@link DiscoveryNode} including all shards
  * that are hosted on that nodes. Each {@link RoutingNode} has a unique node id that can be used to identify the node.
  */
 public class RoutingNode implements Iterable<ShardRouting> {
 
+    static class BucketedShards implements Iterable<ShardRouting> {
+        private static Map<Boolean, Integer> map = new HashMap<Boolean, Integer>() {
+            {
+                put(true, 0);
+                put(false, 1);
+            }
+        };
```

Comment:
is the only use case for indexing into the shards array (0 or 1), will enum be better?

Reply:
Maybe I am missing something here, but enum will define new type which is different than returned from shardRouting.primary() boolean?

Reply:
you are right enum wont work here. Can leave it as is.